extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/service/IdsClearingHouseServiceImpl.java

```diff
@@ -46,20 +51,22 @@ public IdsClearingHouseServiceImpl(
             Hostname hostname,
             URL clearingHouseLogUrl,
             ContractNegotiationStore contractNegotiationStore,
+            TransferProcessStore transferProcessStore,
             Monitor monitor) {
         this.dispatcherRegistry = dispatcherRegistry;
         this.connectorServiceSettings = connectorServiceSettings;
         this.hostname = hostname;
         this.clearingHouseLogUrl = clearingHouseLogUrl;
         this.contractNegotiationStore = contractNegotiationStore;
+        this.transferProcessStore = transferProcessStore;
         this.monitor = monitor;
     }
 
     @Override
     public void logContractAgreement(ContractAgreement contractAgreement, URL clearingHouseLogUrl) {
         monitor.info("Logging contract agreement to ClearingHouse");
         try {
-            var connectorBaseUrl = new URI(String.format("http://%s/", hostname.get()));
+            var connectorBaseUrl = getConnectorBaseUrl();
```

Comment:
Idea: set in constructor and replace hostname with connectorBaseUrl in the fields

Reply:
The Exception can be caught in the constructor and wrapped in an EDCException. In this way misconfiguration can already be identified on startup